From: David Milburn <dmilburn@redhat.com>
Date: Fri, 17 Apr 2009 17:04:14 -0500
Subject: [ata] sata_mv: fix chip type for RocketRaid 1740/1742
Message-id: 20090417220414.GA18982@dhcp-210.hsv.redhat.com
O-Subject: [RHEL5.4 PATCH] sata_mv: fix chip type for Highpoint RocketRaid 1740/1742
Bugzilla: 496338
RH-Acked-by: Jeff Garzik <jgarzik@redhat.com>
RH-Acked-by: Chuck Ebbert <cebbert@redhat.com>

This patch adds the RocketRaid 1720 PCI ID and corrects
the chip type for the 1740 and 1742 PCI cards. This patch
is for BZ 496338, I have only compile tested and it corresponds
to these upstream commits:

commit 46c5784c8fa736c2bb42fe681189b86e99abdc2e
commit 4462254ac6be9150aae87d54d388fc348d6fcead

Please review and ACK.

Thanks,
David

 drivers/ata/sata_mv.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/ata/sata_mv.c b/drivers/ata/sata_mv.c
index c5d561e..36a2090 100644
--- a/drivers/ata/sata_mv.c
+++ b/drivers/ata/sata_mv.c
@@ -663,8 +663,9 @@ static const struct pci_device_id mv_pci_tbl[] = {
 	{ PCI_VDEVICE(MARVELL, 0x5080), chip_5080 },
 	{ PCI_VDEVICE(MARVELL, 0x5081), chip_508x },
 	/* RocketRAID 1740/174x have different identifiers */
-	{ PCI_VDEVICE(TTI, 0x1740), chip_508x },
-	{ PCI_VDEVICE(TTI, 0x1742), chip_508x },
+	{ PCI_VDEVICE(TTI, 0x1720), chip_6042 },
+	{ PCI_VDEVICE(TTI, 0x1740), chip_6042 },
+	{ PCI_VDEVICE(TTI, 0x1742), chip_6042 },
 
 	{ PCI_VDEVICE(MARVELL, 0x6040), chip_604x },
 	{ PCI_VDEVICE(MARVELL, 0x6041), chip_604x },
